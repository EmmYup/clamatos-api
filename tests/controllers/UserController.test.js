describe('UserController', () => {
  beforeAll(() =>
    Promise.all([
      User.create({ name: 'user1', password: '12345', type: 'admin' }),
      User.create({ name: 'user2', password: '12345', type: 'employee' }),
      User.create({ name: 'user3', password: '12345', type: 'employee' }),
    ])
  );
  describe('index', () => {
    it('should find employee users', () =>
      app
        .get('/user')
        .expect(200)
        .then(({ body }) => expect(body).toMatchSnapshot()));
  });
  describe('index', () => {
    it('should find users by name if it is provided', () =>
      app
        .get('/user?name=user3')
        .expect(200)
        .then(({ body }) => expect(body).toMatchSnapshot()));
  });
  describe('show', () => {
    it('should return a single user', () =>
      app
        .get('/user/1')
        .expect(200)
        .then(({ body }) => expect(body).toMatchSnapshot()));
  });
  describe('create', () => {
    it('should create a user', () =>
      app
        .post('/user?name=user4&password=12345&type=employee')
        .expect(201)
        .then(({ body }) => expect(body).toMatchSnapshot()));
  });
  describe('remove', () => {
    it('should delete a user', () => app.delete('/user/2').expect(200));
  });
});
