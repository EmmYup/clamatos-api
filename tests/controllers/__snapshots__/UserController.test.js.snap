// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserController create should create a user 1`] = `
Object {
  "createdAt": "2017-01-01T05:00:00.000Z",
  "id": 4,
  "name": "user4",
  "type": "employee",
  "updatedAt": "2017-01-01T05:00:00.000Z",
}
`;

exports[`UserController index should find employee users 1`] = `
Array [
  Object {
    "createdAt": "2017-01-01T05:00:00.000Z",
    "id": 2,
    "name": "user2",
    "type": "employee",
    "updatedAt": "2017-01-01T05:00:00.000Z",
  },
  Object {
    "createdAt": "2017-01-01T05:00:00.000Z",
    "id": 3,
    "name": "user3",
    "type": "employee",
    "updatedAt": "2017-01-01T05:00:00.000Z",
  },
]
`;

exports[`UserController index should find users by name if it is provided 1`] = `
Array [
  Object {
    "createdAt": "2017-01-01T05:00:00.000Z",
    "id": 3,
    "name": "user3",
    "type": "employee",
    "updatedAt": "2017-01-01T05:00:00.000Z",
  },
]
`;

exports[`UserController show should return a single user 1`] = `
Object {
  "createdAt": "2017-01-01T05:00:00.000Z",
  "id": 1,
  "name": "user1",
  "type": "admin",
  "updatedAt": "2017-01-01T05:00:00.000Z",
}
`;
